perl/modules/Zulip/lib/BarnOwl/Module/Zulip.pm

```diff
@@ -0,0 +1,522 @@
+use strict;
+use warnings;
+
+package BarnOwl::Module::Zulip;
+
+our $VERSION=0.1;
+our $queue_id;
+our $last_event_id;
+our %cfg;
+our $max_retries = 1000;
+our $retry_timer;
+our $tls_ctx;
+our %msg_id_map;
+our $presence_timer;
+
+use AnyEvent;
+use AnyEvent::HTTP;
+use AnyEvent::TLS;
+use JSON;
+use MIME::Base64;
+use URI;
+use URI::Encode;
+use BarnOwl::Hooks;
+use BarnOwl::Message::Zulip;
+use HTTP::Request::Common;
+use Getopt::Long qw(GetOptionsFromArray);
+
+sub fail {
+    my $msg = shift;
+    undef %cfg;
+    undef $queue_id;
+    undef $last_event_id;
+
+    BarnOwl::admin_message('Zulip Error', $msg);
+    die("Zulip Error: $msg\n");
+}
+
+
+sub initialize {
+    my $conffile = BarnOwl::get_config_dir() . "/zulip";
+
+    if (open(my $fh, "<", "$conffile")) {
+        read_config($fh);
+        close($fh);
+    }
+    if ($cfg{'api_url'} =~ /^https/) {
```

Comment:
`Use of uninitialized value $BarnOwl::Module::Zulip::cfg{"api_url"} in pattern match (m//) at barnowl/perl/modules/Zulip/lib/BarnOwl/Module/Zulip.pm line 46.`
(for non-Zulip users)